pjlib/src/pj/string.c

```diff
@@ -310,7 +310,7 @@ PJ_DEF(pj_status_t) pj_strtol2(const pj_str_t *str, long *value)
         return PJ_ETOOSMALL;
     }
 
-    *value = is_negative ? -retval : retval;
+    *value = is_negative ? -(long)retval : retval;
```

Comment:
This will cause regression of #3315 and #3426.

Reply:
Ah right, tried a different approach in the new commit.

Reply:
Perhaps just changing it to `retval > PJ_MAXLONG` is enough, as described in #3315?
This way, no additional if block is needed.

Reply:
Right, thought about that, but it will change behavior (i.e: `PJ_MINLONG` will no longer accepted) which generally should be avoid in this kind of patch, so is changing behavior acceptable here?

Reply:
Ah I see.
How about the original patch, i.e. (long)-retval? Will it still trigger warning?

Reply:
Same warning as before this patch: `warning C4146: unary minus operator applied to unsigned type, result still unsigned`